### PR TITLE
remove agemo specifics from config_utils

### PR DESCRIPTION
**Pull request by devkelley**
Closes / Fixes / Resolves #0000

## Description

  Removes agemo specific implementation from config_utils file.
  Splits up the gathering of configuration into:
  - default config
  - config from file
  - commandline config

This separation should allow for the util code to be easily placed in a common repo in the future. It should also allow for more flexibility.

### CODE REVIEW

File: common/src/config_utils.rs [hunk 1]
```diff
@@ -2,83 +2,467 @@
...
+pub fn load_default_config_from_file(
+    config_file: &ConfigFileMetadata,
+    default_dir: &Dir,
+) -> Result<Box<dyn Source + Send + Sync>, Box<dyn std::error::Error + Send + Sync>> {
+    // Get appropriate default config.
+    let file = default_dir
+        .get_file(&config_file.name)
+        .ok_or(std::io::Error::new(
+            std::io::ErrorKind::NotFound,
+            format!("Unable to find default file '{}'.", &config_file.name),
```
Comment (ashbeitz) [inline common/src/config_utils.rs:106]: Nit: Be consistent with your capitalization of the first word of the error message.  You have previously, started with a capital letter.

File: common/src/config_utils.rs [hunk 2]
```diff
@@ -2,83 +2,467 @@
...
-    let overrides_config_file_path = config_path.join(overrides_file);
+    Ok(config_path)
+}
+
+/// Read config from a configuration file located at the given config path at runtime.
+///
+/// # Arguments
+/// * `config_file` - The config file to read.
+/// * `config_path` - The path to the directory containing the config.
+pub fn read_from_file<TPath>(
```
Comment (ashbeitz) [inline common/src/config_utils.rs:156]: Perhaps rename "read_config_from_file".  This makes it consistent with the next function.

File: common/src/config_utils.rs [hunk 3]
```diff
@@ -2,83 +2,467 @@
...
+    default_config_file: &ConfigFileMetadata,
+    default_dir: &Dir,
+    svc_home_metadata: &SvcConfigHomeMetadata,
+    cmdline_args: Option<TArgs>,
+) -> Result<TConfig, Box<dyn std::error::Error + Send + Sync>>
+where
+    TConfig: for<'de> serde::Deserialize<'de>,
+    TArgs: Source + Send + Sync,
+{
+    // Load default configuration for the given configuration file.
```
Comment (ashbeitz) [inline common/src/config_utils.rs:218]: Nit: I feel that some of the comments in this function are not necessary, as you have used great function names that clearly identify what is being done,

File: pub-sub-service/src/load_config.rs [hunk 4]
```diff
@@ -78,19 +94,37 @@ pub struct Settings {
...
 
-/// Load a configuration file.
+/// Load configuration given a file and commandline arguments.
 ///
 /// # Arguments
 /// * `config_file_name` - Name of the config file to load settings from.
 /// * `args` - Optional commandline config arguments.
 pub fn load_config<T>(
     config_file_name: &str,
+    default_file_name: &str,
```
Comment (ashbeitz) [inline pub-sub-service/src/load_config.rs:104]: Please add the details on this new parameter to the method's documentation.

File: common/src/config_utils.rs (hunk 1 above)
Comment (devkelley) [reply common/src/config_utils.rs:106]: Standardized

File: common/src/config_utils.rs (hunk 2 above)
Comment (devkelley) [reply common/src/config_utils.rs]: renamed

File: common/src/config_utils.rs (hunk 3 above)
Comment (devkelley) [reply common/src/config_utils.rs]: removed some comments

File: pub-sub-service/src/load_config.rs (hunk 4 above)
Comment (devkelley) [reply pub-sub-service/src/load_config.rs:105]: Done

**Comment by devkelley** [comment]
Need to reopen this pr against main

**Pull request closed without merging**